spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDeploymentController.java

```diff
@@ -63,11 +65,13 @@ public TaskDeploymentController(TaskService taskService) {
 	 * @param taskName the name of the existing task to be executed (required)
 	 * @param properties the runtime properties for the task, as a comma-delimited list of
 	 * 					 key=value pairs
+	 * @param params the runtime commandline arguments
 	 */
 	@RequestMapping(value = "/{name}", method = RequestMethod.POST)
 	@ResponseStatus(HttpStatus.CREATED)
-	public void deploy(@PathVariable("name") String taskName, @RequestParam(required = false) String properties) {
-		this.taskService.executeTask(taskName, DeploymentPropertiesUtils.parse(properties));
+	public void deploy(@PathVariable("name") String taskName, @RequestParam(required = false) String properties,
+			@RequestParam(required = false) List<String> params) {
+		this.taskService.executeTask(taskName, DeploymentPropertiesUtils.parse(properties), params);
```

Comment:
probably need to parse each of the parameters (delimited by space) out into a list.   Else it will send the all params as a single param instead of collection parameters.  
```
		List<String> paramsToUse = new ArrayList<>();
		if(params != null) {
			for (String paramsItem : params) {
				for (String param : StringUtils.delimitedListToStringArray(paramsItem, " ")) {
					paramsToUse.add(param);
				}
			}
		}
		this.taskService.executeTask(taskName, DeploymentPropertiesUtils.parse(properties), paramsToUse);
```